### PR TITLE
Simplify aggs collector methods in SearchContext

### PR DESCRIPTION
**Pull request by javanna**
SearchContext exposes a method that returns a modifiable map of collectors, keyed by a generic object. In practice, we use this method to register a single collector for aggregations, and the key is always `QueryPhase`. Also, the logic is hard to follow as there is a single method that returns the map which gets modified in the callers.

This commit replaces the existing `queryCollectors` method with two new methods that are specific to aggs, which allow to register a single collector: `registerAggsCollector` and `getAggsCollector`.

### CODE REVIEW

**Review by cbuescher — APPROVED** [review]
Great simplification, LGTM

**Review by martijnvg — APPROVED** [review]
LGTM
